File: Csharp/Medium/LC_2130.cs
namespace Csharp.Medium
{
    public class MaximumTwinSum_2130
    {
        public int PairSum(ListNode head)
        {
            IList<int> list = new List<int>();
            while (head != null)
            {
                list.Add(head.val);
                head = head.next;
            }
            int left = 0, right = list.Count - 1, twinSum = int.MinValue;
            while (left < right)
            {
                twinSum = Math.Max(list[left] + list[right], twinSum);
                ++left;
                --right;
            }
            return twinSum;
        }

        public class ListNode
        {
            public int val;
            public ListNode next;
            public ListNode(int val = 0, ListNode next = null)
            {
                this.val = val;
                this.next = next;
            }
        }
    }
}

File: Python/Medium/LC_2130.py
def pairSum(head) -> int:
    list = []
    while head:
        list.append(head.val)
        head = head.next

    L, R, sum = 0, len(list) - 1, -1
    while L < R:
        sum = max(list[L] + list[R], sum)
        L += 1
        R -= 1
    return sum